pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -492,20 +492,26 @@ func (r *Reconciler) exists() (bool, error) {
 			// this error type bubbles back up to the machine-controller to allow
 			// us to delete machines that were never properly created due to
 			// invalid zone.
-			return false, machinecontroller.InvalidMachineConfiguration(fmt.Sprintf("%s: Machine does not exist", r.machine.Name))
+
+			// If the machine has a node assigned, we know that it actually exists,
+			// but it also has an invalid configuration.
+			if r.machine.Status.NodeRef != nil {
+				return true, true, machinecontroller.InvalidMachineConfiguration(fmt.Sprintf("%s: Zone does not exist", r.providerSpec.Zone))
```

Comment:
I don't think we need this bool, you can unwrap/check that the error return type is an `invalidMachineConfiguration` within the parent scope, this would be more idiomatic